cmd/docker/docker.go

```diff
@@ -247,11 +246,18 @@ func tryPluginRun(dockerCli command.Cli, cmd *cobra.Command, subcommand string,
 				// receive signals due to sharing a pgid with the parent CLI
 				continue
 			}
-			if conn != nil {
-				if err := conn.Close(); err != nil {
-					_, _ = fmt.Fprintf(dockerCli.Err(), "failed to signal plugin to close: %v\n", err)
+			select {
+			// connChan will close itself once we receive the connection
+			// thus further loops will not block on connChan
+			case c := <-connChan:
+				if c != nil {
+					if err := c.Close(); err != nil {
+						_, _ = fmt.Fprintf(dockerCli.Err(), "failed to signal plugin to close: %v\n", err)
+					}
 				}
-				conn = nil
+			// timeout here if for some reason the channel is still open and not populated.
+			case <-time.After(1 * time.Millisecond):
```

Comment:
Are we sure we want to timeout here after 1 ms with no errors at all? I assume you chose this instead of `default` to allow some extra time for the response to arrive on the channel, but what are we expecting to happen in those cases?

Reply:
We don't throw an error here since the function will continue executing and call the fallback https://github.com/docker/cli/blob/ad778cf743c1876098671035bc183aa39762827d/cmd/docker/docker.go#L265

but you're right, we could also just do a `default` and continue. The timeout is arbitrary since we only execute this part of the code when the user terminates the CLI, by that time the channel should be populated and if not, just gracefully continue.

Reply:
Let's prefer `default:` here as that is the idiomatic pattern for non-blocking channel reads.